### ISSUE TITLE
Struct accessors using the wrong name for the type in the OVERLAPPED struct

### ISSUE THREAD

**Issue by sotteson1** [microsoft/ClangSharp#217]
I just synced to the latest source and built it and I'm trying to use it for the Win32metdata project. It looks like something has changed that has broken the scraping of the _OVERLAPPED struct:

The accessor properties for Offset and OffsetHigh use _Anonymous_e__Struct._Anonymous_e__Struct instead of _Anonymous_e__Union._Anonymous_e__Struct. The previous version didn't have this problem. This keeps it from compiling.

```C#
    public unsafe partial struct OVERLAPPED
    {
        [NativeTypeName("ULONG_PTR")]
        public UIntPtr Internal;

        [NativeTypeName("ULONG_PTR")]
        public UIntPtr InternalHigh;

        [NativeTypeName("_OVERLAPPED::(anonymous union at D:/repos/win32metadata/artifacts/InstalledPackages/Microsoft.Windows.SDK.CPP.10.0.19041.5/c/Include/10.0.19041.0/um/minwinbase.h:55:5)")]
        public _Anonymous_e__Union Anonymous;

        [NativeTypeName("HANDLE")]
        public IntPtr hEvent;

        public ref uint Offset
        {
            get
            {
                fixed (_Anonymous_e__Struct._Anonymous_e__Struct* pField = &Anonymous.Anonymous)
                {
                    return ref pField->Offset;
                }
            }
        }

        public ref uint OffsetHigh
        {
            get
            {
                fixed (_Anonymous_e__Struct._Anonymous_e__Struct* pField = &Anonymous.Anonymous)
                {
                    return ref pField->OffsetHigh;
                }
            }
        }

        public ref void* Pointer
        {
            get
            {
                fixed (_Anonymous_e__Union* pField = &Anonymous)
                {
                    return ref pField->Pointer;
                }
            }
        }

        [StructLayout(LayoutKind.Explicit)]
        public unsafe partial struct _Anonymous_e__Union
        {
            [FieldOffset(0)]
            [NativeTypeName("_OVERLAPPED::(anonymous struct at D:/repos/win32metadata/artifacts/InstalledPackages/Microsoft.Windows.SDK.CPP.10.0.19041.5/c/Include/10.0.19041.0/um/minwinbase.h:56:9)")]
            public _Anonymous_e__Struct Anonymous;

            [FieldOffset(0)]
            [NativeTypeName("PVOID")]
            public void* Pointer;

            public partial struct _Anonymous_e__Struct
            {
                [NativeTypeName("DWORD")]
                public uint Offset;

                [NativeTypeName("DWORD")]
                public uint OffsetHigh;
            }
        }
    }
```

**Comment by tannergooding** [comment]
Taking a look...

**Comment by tannergooding** [comment]
@sotteson1, this will be resolved with #219.

The logic was simplified and moved to be part of `IndirectFieldDecl` traversal (now that libClangSharp correctly exposes those) but it was accidentally reusing the the anonymous record decl when generating the type name rather than using the parent record decl.

#219 resolves things by ensuring the correct record decl is used when constructing the type to use.
